src/tss2-tcti/tctildr-dl.c

```diff
@@ -91,36 +91,45 @@ handle_from_name(const char *file,
     } else {
         LOG_DEBUG("Could not load TCTI file: \"%s\": %s", file, dlerror());
     }
+
+    file_xfrm = calloc(PATH_MAX, sizeof(char));
+    if (file_xfrm == NULL) {
+        return TSS2_TCTI_RC_MEMORY;
+    }
     /* 'name' alone didn't work, try libtss2-tcti-<name>.so.0 */
     size = snprintf(file_xfrm,
-                    sizeof (file_xfrm),
+                    PATH_MAX,
                     TCTI_NAME_TEMPLATE_0,
                     file);
```

Comment:
To be less greedy on memory why not run snprintf twice:
1. With a NULL buffer and 0 buffer size to get needed size
2. bound check it to path max
3. calloc the buffer
4. call snprintf with buffer.

Reply:
> To be less greedy on memory why not run snprintf twice:
> 
> 1. With a NULL buffer and 0 buffer size to get needed size
> 2. bound check it to path max
> 3. calloc the buffer
> 4. call snprintf with buffer.

Thanks for your review. I've revised the code according to your comments.